.circleci/config.yml

```diff
@@ -77,17 +85,24 @@ jobs:
       - run:
           name: Run client tests
           command: 'docker run -e CI=true ${DOCKER_USER}/client-test${CIRCLE_BUILD_NUM} npm run test -- --coverage'
+      - run:
+          name: 'Skip the next steps for PRs'
+          command: |
+            # Check the environment and gracefully stop the job for PRs
+            if [[ "$CIRCLE_BRANCH" != "master" ]] && [[ "$CIRCLE_BRANCH" != "develop" ]] ; then
+              circleci-agent step halt
+            fi
```

Comment:
Is there no way to make steps conditional, instead of force stopping the build early?

Reply:
Conditional steps configuration is very limited, and we cannot split the job in two because we need to reuse the local docker cache. I cannot see an easy alternative way of doing it.